src/v1/device_manager_client.ts

```diff
@@ -107,6 +108,28 @@ interface GetRegistryCredentialsResponse {
   url: string;
 }
 
+function isBinaryDataFormat(): boolean {
+  const binaryDataFormat = process.env.BINARYDATA_AND_TIME_GOOGLE_FORMAT;
+  if (
+    typeof binaryDataFormat !== 'undefined' &&
+    binaryDataFormat.toLowerCase() === 'true'
+  ) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function timeSecondsNanos(data: string): protos.google.protobuf.ITimestamp {
+  const timeStamp: protos.google.protobuf.ITimestamp = {};
+  const time = Timestamp.fromString(data);
```

Comment:
Minor nitpick: the 'const time = Timestamp...' should be inside the following 'if (data) {...' block. I'll approve now.